support-lib/wasm/djinni_wasm.cpp

```diff
@@ -201,10 +189,22 @@ EM_JS(void, djinni_register_name_in_ns, (const char* prefixedName, const char* n
         ns[name] = Module[prefixedName];
 });
 
+void djinni_throw_native_exception(const std::exception& e) {
+    if (const auto* jsEx = dynamic_cast<const JsException*>(&e)) {
```

Comment:
@LiFengSC By the way, is this our first usage of RTTI in djinni? If it is, is this worth taking the dependency on it? We don't need RTTI for exceptions.

Could we generate code that catches both types, without using rtti?

```
    } catch(const JsException& e) {
      // re-throw JS
    } catch(const std::exception& e) {
        // re-throw native
    }
```

Reply:
Actually I guess matching the catch clause requires RTTI anyways? Just wondering here really

Reply:
There are other uses of RTTI, eg. here we use it to detect whether the interface is a JS object or C++ object.
```
    static em::val _toJs(const std::shared_ptr<I>& c) {
        if (c == nullptr) {
            // null object
            return em::val::undefined();
        }
        else if (auto* p = dynamic_cast<JsProxyBase*>(c.get())) {
            // unwrap existing js proxy
            return p->_jsRef();
        } else {
            return GetOrCreateCppProxy<Self, void>()(c);
        }
    }
```

Reply:
We should develop a better understanding of the size and perf overhead we add by relying on RTTI CC @bholmes-sc